clippy_lints/src/implicit_saturating_sub.rs

```diff
@@ -99,17 +105,31 @@ impl<'tcx> LateLintPass<'tcx> for ImplicitSaturatingSub {
                             };
                         }
                     },
-                    ExprKind::Path(ref cond_num_path) => {
-                        if INT_TYPES.iter().any(|int_type| match_qpath(cond_num_path, &[int_type, "MIN"])) {
-                            print_lint_and_sugg(cx, &var_name, expr);
-                        };
+                    ExprKind::Path(QPath::TypeRelative(ty, name)) => if name.ident.as_str() == "MIN" {
+                        let ty = cx.typeck_results().node_type(ty.hir_id);
+                        if INT_TYPES.iter()
+                            .filter_map(|&ty| cx.tcx.lang_items().require(ty).ok())
+                            .map(|id| cx.tcx.type_of(id))
```

Comment:
Did you try comparing to `impl_of_method`? That would be a more direct approach since the impl is the lang item.

Reply:
Just now, yes. Looks about as legible.